src/ledger/LedgerManagerImpl.cpp

```diff
@@ -623,9 +624,23 @@ LedgerManagerImpl::startCatchup(CatchupConfiguration configuration,
         throw std::invalid_argument("Target ledger is not newer than LCL");
     }
 
-    setCatchupState(CatchupState::APPLYING_HISTORY);
-    assert(manualCatchup == mSyncingLedgers.empty());
+    if (manualCatchup)
+    {
+        if (getState() == LM_CATCHING_UP_STATE &&
+            mCatchupState != CatchupState::WAITING_FOR_CLOSING_LEDGER)
+        {
+            throw std::runtime_error(
+                "Unable to perform second catchup before first one finishes");
+        }
+        assert(mSyncingLedgers.empty());
+        mSyncingLedgers.reset();
+    }
+    else
+    {
+        assert(!mSyncingLedgers.empty());
```

Comment:
again, I don't like this coupling with `mSyncingLedgers` and all those asserts.
Why do we need a non empty `mSyncingLedgers` at this stage? Expectations wrt `mSyncingLedgers` should be very simple. If you need an extra state, just add it.

Isn't it possible to have `addToSyncingLedgers` not actually add anything to `mSyncingLedgers` (like if `reset` was not called, which would potentially trip those asserts?). Right now it's hard to tell if this would just work.

Reply:
That would mean that startup is starting in automatic mode without any externalized ledger received by stellar-core, that is clearly wrong (only manual catchup can start that way). Also at that point `
addToSyncingLedgers` must add anything, because `reset` being not called before is not possible (that would some previous catchup is ongoing, which mean startCatchup is not possible to execute).